File: Python/Easy/1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        answer = count = 0
        for c in s:
            if c == "L":
                count += 1
            else:
                count -= 1
            if count == 0:
                answer += 1

        return answer

File: Java/Easy/1221.java
class Solution {
    public int balancedStringSplit(String s) {
        // greedy approach
        int count = 0;
        int holder = 0; // count number of right the moment it is 0 add to count
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'R') {
                holder++;
            } else {
                holder--;
            }
            if (holder == 0) {
                count++;
            }
        }
        return count;
    }
}